responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py

```diff
@@ -153,6 +154,87 @@ def test_counterfactual_manager_save_load(self, tmpdir):
         assert counterfactual_config_list[0].explainer is not None
         assert counterfactual_config_list[1].explainer is not None
 
+    @pytest.mark.parametrize('feature_importance', [True, False])
+    def test_counterfactual_manager_request_counterfactuals(
+            self, feature_importance):
+        X_train, X_test, y_train, y_test, feature_names, _ = \
+            create_iris_data()
+
+        model = create_lightgbm_classifier(X_train, y_train)
+        X_train['target'] = y_train
+        X_test['target'] = y_test
+
+        rai_insights = RAIInsights(
+            model=model,
+            train=X_train,
+            test=X_test.iloc[0:10],
+            target_column='target',
+            task_type='classification')
+
+        rai_insights.counterfactual.add(
+            total_CFs=10, desired_class=2,
+            features_to_vary=[feature_names[0]],
+            permitted_range={feature_names[0]: [2.0, 5.0]},
+            feature_importance=feature_importance)
+        rai_insights.counterfactual.add(
+            total_CFs=10, desired_class=1,
+            features_to_vary=[feature_names[0]],
+            permitted_range={feature_names[0]: [2.0, 5.0]},
+            feature_importance=feature_importance)
+        rai_insights.counterfactual.compute()
+
+        test_instance = X_test.iloc[0:1].drop('target', axis=1)
+        query_id = \
+            rai_insights.counterfactual._counterfactual_config_list[0].id
+        counterfactual_obj_serialized = \
+            rai_insights.counterfactual.request_counterfactuals(
+                query_id, test_instance)
+
+        assert counterfactual_obj_serialized['test_data'] is not None
+        assert counterfactual_obj_serialized['cfs_list'] is not None
+        if feature_importance:
+            assert counterfactual_obj_serialized[
+                'local_importance'] is not None
+        else:
+            assert counterfactual_obj_serialized[
+                'local_importance'] is None
+        assert counterfactual_obj_serialized['summary_importance'] is None
```

Comment:
nit: maybe you can loop through these as well - have a list of None and not None fields and assert them in a loop

Reply:
Done.